pageserver/src/tenant/delete.rs

```diff
@@ -361,25 +361,17 @@ impl DeleteTenantFlow {
         conf: &'static PageServerConf,
         remote_mark_exists: bool,
         tenant: &Tenant,
-    ) -> Result<Option<DeletionGuard>, DeleteTenantError> {
-        let acquire = |t: &Tenant| {
+    ) -> Option<DeletionGuard> {
+        let tenant_id = tenant.tenant_id;
+
+        if remote_mark_exists || conf.tenant_deleted_mark_file_path(&tenant_id).exists() {
```

Comment:
NB: the `.exists()` actually masks falliblity of the `stat` system call. Should be using `try_exists` + panic on any error other than NotExist

Reply:
I don't think we should panic unless the plan is to create a lot more work in future for "epic: graceful handling of errors", otherwise a good point which might explain why this was fallible in the first place.